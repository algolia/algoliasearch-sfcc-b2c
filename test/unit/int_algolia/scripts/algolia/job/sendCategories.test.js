const GlobalMock = require('../../../../../mocks/global');
const CategoryMock = require('../../../../../mocks/dw/catalog/Category');

global.request = new GlobalMock.RequestMock();

jest.mock('dw/catalog/CatalogMgr', () => {}, {virtual: true});
jest.mock('dw/system/Site', () => {}, {virtual: true});

jest.mock('dw/web/URLUtils', () => {
    return {
        https: function(endpoint, param, id) {
            var relURL = '/on/demandware.store/Sites-Algolia_SFRA-Site/';
            return relURL + global.request.getLocale() + '/' + endpoint + '?' + param + '=' + id;
        },
    }
}, {virtual: true});
jest.mock('*/cartridge/scripts/algolia/model/algoliaLocalizedCategory', () => {
    return jest.requireActual('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/model/algoliaLocalizedCategory');
}, {virtual: true});

const job = require('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/job/sendCategories');

describe('getSubCategoriesModels', () => {
    // We define categories mocks that represent the following categories hierarchy:
    // Electronics
    // |__Digital Cameras
    // |__Audio
    //    |__Headphones
    const category = new CategoryMock({ name: 'Electronics' });
    const subcategory1 = new CategoryMock({ name: 'Digital Cameras', parent: category });
    const subcategory2 = new CategoryMock({ name: 'Audio', parent: category });
    const subsubcategory2 = new CategoryMock({ name: 'Headphones', parent: subcategory2 });
    subcategory2.subcategories = [subsubcategory2];
    // This category won't be exported as showInMenu is false
    const subcategory3 = new CategoryMock({ name: 'Video', parent: category, showInMenu: false });
    category.subcategories = [subcategory1, subcategory2, subcategory3];
    const catalogId = 'testCatalog';

    test('default locale', () => {
        var categoriesModels = job.getSubCategoriesModels(category, catalogId, 'default');
        expect(categoriesModels).toMatchSnapshot();
    });
    test('french locale', () => {
        var categoriesModels = job.getSubCategoriesModels(category, catalogId, 'fr');
        expect(categoriesModels).toMatchSnapshot();
    });
});
