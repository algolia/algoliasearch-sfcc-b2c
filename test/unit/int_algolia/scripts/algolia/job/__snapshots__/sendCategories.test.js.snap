// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSubCategoriesModels default locale 1`] = `
[
  algoliaLocalizedCategory {
    "_tags": [
      "id:testCatalog/storefront-catalog-m-en/electronics",
    ],
    "description": "Description of Electronics",
    "id": "testCatalog/storefront-catalog-m-en/electronics",
    "image": "http://example.com/electronics.jpg",
    "name": "Electronics",
    "objectID": "testCatalog/storefront-catalog-m-en/electronics",
    "subCategories": [
      "testCatalog/storefront-catalog-m-en/digital-cameras",
      "testCatalog/storefront-catalog-m-en/audio",
    ],
    "thumbnail": "http://example.com/electronics-thumbnail.jpg",
    "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/default/Search-Show?cgid=storefront-catalog-m-en/electronics",
  },
  algoliaLocalizedCategory {
    "_tags": [
      "id:testCatalog/storefront-catalog-m-en/digital-cameras",
    ],
    "description": "Description of Digital Cameras",
    "id": "testCatalog/storefront-catalog-m-en/digital-cameras",
    "image": "http://example.com/digital-cameras.jpg",
    "name": "Digital Cameras",
    "objectID": "testCatalog/storefront-catalog-m-en/digital-cameras",
    "parent_category_id": "testCatalog/storefront-catalog-m-en/electronics",
    "thumbnail": "http://example.com/digital-cameras-thumbnail.jpg",
    "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/default/Search-Show?cgid=storefront-catalog-m-en/digital-cameras",
  },
  algoliaLocalizedCategory {
    "_tags": [
      "id:testCatalog/storefront-catalog-m-en/audio",
    ],
    "description": "Description of Audio",
    "id": "testCatalog/storefront-catalog-m-en/audio",
    "image": "http://example.com/audio.jpg",
    "name": "Audio",
    "objectID": "testCatalog/storefront-catalog-m-en/audio",
    "parent_category_id": "testCatalog/storefront-catalog-m-en/electronics",
    "subCategories": [
      "testCatalog/storefront-catalog-m-en/headphones",
    ],
    "thumbnail": "http://example.com/audio-thumbnail.jpg",
    "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/default/Search-Show?cgid=storefront-catalog-m-en/audio",
  },
  algoliaLocalizedCategory {
    "_tags": [
      "id:testCatalog/storefront-catalog-m-en/headphones",
    ],
    "description": "Description of Headphones",
    "id": "testCatalog/storefront-catalog-m-en/headphones",
    "image": "http://example.com/headphones.jpg",
    "name": "Headphones",
    "objectID": "testCatalog/storefront-catalog-m-en/headphones",
    "parent_category_id": "testCatalog/storefront-catalog-m-en/audio",
    "thumbnail": "http://example.com/headphones-thumbnail.jpg",
    "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/default/Search-Show?cgid=storefront-catalog-m-en/headphones",
  },
]
`;

exports[`getSubCategoriesModels french locale 1`] = `
[
  algoliaLocalizedCategory {
    "_tags": [
      "id:testCatalog/storefront-catalog-m-en/electronics",
    ],
    "description": "Description française de Electronics",
    "id": "testCatalog/storefront-catalog-m-en/electronics",
    "image": "http://example.com/electronics.jpg",
    "name": "Nom français de Electronics",
    "objectID": "testCatalog/storefront-catalog-m-en/electronics",
    "subCategories": [
      "testCatalog/storefront-catalog-m-en/digital-cameras",
      "testCatalog/storefront-catalog-m-en/audio",
    ],
    "thumbnail": "http://example.com/electronics-thumbnail.jpg",
    "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/fr/Search-Show?cgid=storefront-catalog-m-en/electronics",
  },
  algoliaLocalizedCategory {
    "_tags": [
      "id:testCatalog/storefront-catalog-m-en/digital-cameras",
    ],
    "description": "Description française de Digital Cameras",
    "id": "testCatalog/storefront-catalog-m-en/digital-cameras",
    "image": "http://example.com/digital-cameras.jpg",
    "name": "Nom français de Digital Cameras",
    "objectID": "testCatalog/storefront-catalog-m-en/digital-cameras",
    "parent_category_id": "testCatalog/storefront-catalog-m-en/electronics",
    "thumbnail": "http://example.com/digital-cameras-thumbnail.jpg",
    "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/fr/Search-Show?cgid=storefront-catalog-m-en/digital-cameras",
  },
  algoliaLocalizedCategory {
    "_tags": [
      "id:testCatalog/storefront-catalog-m-en/audio",
    ],
    "description": "Description française de Audio",
    "id": "testCatalog/storefront-catalog-m-en/audio",
    "image": "http://example.com/audio.jpg",
    "name": "Nom français de Audio",
    "objectID": "testCatalog/storefront-catalog-m-en/audio",
    "parent_category_id": "testCatalog/storefront-catalog-m-en/electronics",
    "subCategories": [
      "testCatalog/storefront-catalog-m-en/headphones",
    ],
    "thumbnail": "http://example.com/audio-thumbnail.jpg",
    "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/fr/Search-Show?cgid=storefront-catalog-m-en/audio",
  },
  algoliaLocalizedCategory {
    "_tags": [
      "id:testCatalog/storefront-catalog-m-en/headphones",
    ],
    "description": "Description française de Headphones",
    "id": "testCatalog/storefront-catalog-m-en/headphones",
    "image": "http://example.com/headphones.jpg",
    "name": "Nom français de Headphones",
    "objectID": "testCatalog/storefront-catalog-m-en/headphones",
    "parent_category_id": "testCatalog/storefront-catalog-m-en/audio",
    "thumbnail": "http://example.com/headphones-thumbnail.jpg",
    "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/fr/Search-Show?cgid=storefront-catalog-m-en/headphones",
  },
]
`;
